hir/src/module.rs

```diff
@@ -515,6 +515,10 @@ impl<'m> ModuleFunctionBuilder<'m> {
         self.function.id
     }
 
+    pub fn func_builder(&mut self) -> FunctionBuilder {
```

Comment:
@bitwalker I added `FunctionBuilder` construction here since I heavily use it throughout the SSA form building and op translation.

Reply:
That's fine, we may want a trait for this actually, since the `ModuleFunctionBuilder` and `FunctionBuilder` should be basically identical, just used in slightly different circumstances (the former is really just for initial construction of a function as part of a module, the latter is a more general utility for constructing/modifying functions, but they should have essentially the same API).

Reply:
Got it. 
Speaking of "function builders". As I mentioned in the PR description,  I created `FunctionBuilderExt` as a wrapper around Miden IR FunctionBuilder and SSABuilder, providing additional API for dealing with mutable variables and SSA construction. I think of it as `FunctionBuilder` with SSA building capabilities. `FunctionBuilderExt` and `SSABuilder` do not have anything Wasm-specific and can be merged into the unified function-building API.

Reply:
Definitely, those would be welcome additions to the builders! A lot of what is there now is just the bare minimum needed to represent the various instructions, as well as enough additional utilities for some of the tests we've written thus far, but there is definitely a lot of room for improvement as you've noted.